apps/evm/src/clients/api/queries/getPendingRewards/index.ts

```diff
@@ -114,18 +114,15 @@ const getPendingRewards = async ({
     ...isolatedPoolRewardTokenAddresses,
     ...primeRewardTokenAddresses,
   ]);
-  const rewardTokenPricesPromises = Promise.allSettled(
-    rewardTokenAddresses.map(rewardTokenAddress =>
-      resilientOracleContract.getPrice(rewardTokenAddress),
-    ),
-  );
-
-  const rewardTokenPricesResults = await rewardTokenPricesPromises;
+  const tokenPriceMantissaMapping = await getApiTokenPrice(rewardTokenAddresses);
 
-  const tokenPriceMapping: Record<string, BigNumber> = rewardTokenPricesResults.reduce<{
+  const tokenPriceMapping: Record<string, BigNumber> = Object.entries(
+    tokenPriceMantissaMapping,
+  ).reduce<{
     [address: string]: BigNumber;
-  }>((acc, rewardTokenPricesResult, index) => {
-    const rewardTokenAddress = rewardTokenAddresses[index];
+  }>((acc, tokenPriceMantissaTuple) => {
+    const rewardTokenAddress = tokenPriceMantissaTuple[0];
+    const rewardTokenPriceMantissa = tokenPriceMantissaTuple[1];
     const rewardToken = findTokenByAddress({
```

Comment:
We can remove this check actually. We are sending the list of tokens we want to fetch the prices of, so by default the list is verified. Even if an unlisted token was added to the mapping, it wouldn't have any impact since we don't use the mapping as a list of tokens but just a reference for prices.

Reply:
You mean findTokenByAddress? I need to use it to find the token record and then use its decimals to calculate the USD value.

Reply:
Ah yes, my bad